Makefile.include

```diff
@@ -277,6 +277,12 @@ export PREFIX ?= $(if $(TARGET_ARCH),$(TARGET_ARCH)-)
 # be done in makefile.iotlab.single.inc.mk which is included after.
 ifneq (,$(IOTLAB_NODE))
   PROGRAMMER ?= iotlab
+  # Boards on iotlab are flashed with an ELF, except wsn430
+  ifneq (,$(findstring wsn430,$(BOARD)))
+    FLASHFILE ?= $(HEXFILE)
+  else
+    FLASHFILE ?= $(ELFFILE)
+  endif
```

Comment:
Shouldn't this be preset by the `board`/`cpu`?

Reply:
This is normally done by the board. But since now we set PROGRAMMER by default when using IOTLAB_NODE, the logic is sometimes skipped in the board logic.
I think it's safer to manage this here because this is iotlab related. But @cladmi could have another opinion on that.

Reply:
You would only need to say `FLASHFILE ?= $(ELFFILE)` and keep the `wsn430` thing specific in the board.

However, can't the `makefile.iot-lab.single.inc.mk` be moved before the check ?
I think it should be working now from what I remember.
I just originally put it at the end near the other one as, maybe it was a well thought choice.

I would also even rather manage to move the file before parsing the board `Makefile.include` directly here.
But this would require changing now the `FFLAGS` to `FFLAGS ?=` in all boards, and same for `FLASHER` and all. `RESET` and `RESET_FLAGS` are already defined like this (even if there is no static check to guarantee it) in the same way as https://github.com/RIOT-OS/RIOT/pull/11649

Currently it would mainly require changing the definition for the `esp` to define a `FFLAGS_ESPTOOL` or something that would then be used as `FFLAGS ?= $(FFLAGS_ESPTOOL)`,
and migrate to have `?=` on all the other ones with a static test.
I did not analyze `DEBUG` and in detail the impact when both `jlink` and `openocd` are included.

Would be great as I currently need to define `override FFLAGS` in my test machine configurations in a `RIOT_MAKEFILES_GLOBAL_PRE` files.
I even do it for `RESET` and `TERM` to be safe in as it is not enforced.
https://git.imp.fu-berlin.de/riot-appstore/riot-buildsystem-ci-server/blob/89275d11727b0baa2ab93233c0fb640478e5da01/boards/vaduz-boards.mk.pre#L37-51

Reply:
> You would only need to say FLASHFILE ?= $(ELFFILE) and keep the wsn430 thing specific in the board.

Indeed.

> can't the makefile.iot-lab.single.inc.mk be moved before the check ?

I can try but not before the next 2 days. And I'd like, if possible to keep this PR minimal. I don't think we can fix the problem quickly if we start playing with FFLAGS, ESPTOOL, etc which are not directly related to iotlab support.

Reply:
> > can't the makefile.iot-lab.single.inc.mk be moved before the check ?
> 
> I can try but not before the next 2 days. And I'd like, if possible, to keep this PR minimal. I don't think we can fix the problem quickly if we start playing with FFLAGS, ESPTOOL, etc which are not directly related to iotlab support.

Agreed.

Reply:
It still works if I move the `iotlab.single.inc.mk` file here:
https://github.com/RIOT-OS/RIOT/blob/3e0a3b1746d9b56a0de13b39f6b4fd9fbf7f59ea/Makefile.include#L450

Would it be fine for you @cladmi ?